src/components/directory-content.js

```diff
@@ -949,8 +1011,10 @@ const DirectoryContent = () => {
         }
 
         // If no selection or currentChildren = null (first time) render nothing
-        // TODO : Make a beautiful page here
         if (!currentChildren || !selectedDirectory) {
+            if (treeData.rootDirectories.length === 0 && treeData.initialized) {
```

Comment:
Why do you need the **initialized**? 
**treeData.rootDirectories.length === 0** is not sufficient in order to show renderNoContent?

Reply:
I think is would be cleaner with a "isLoading" functionnality, but it might be a little complex since treeData is is redux
here it sounds like a hack

Reply:
at the beginning I thought about adding is Loading functionality, but didnt find another way to do it since treedata is redux